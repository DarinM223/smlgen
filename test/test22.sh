../smlgen test22.sml no:s --recurmod --test
if cmp --silent test22.sml.test test22.sml.expected; then
  echo "test22 succeeded"
else
  echo "test22 failed"
  echo "Diff:"
  diff test22.sml.test test22.sml.expected
  exit 1
fi