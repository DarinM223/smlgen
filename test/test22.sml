structure Foo = struct
    datatype foo = Foo of { value : foo' option }
    withtype foo' = { opt1 : foo option, opt2 : foo'' option }
    and foo'' = Bar.bar * Foo.foo
end

structure Bar = struct
    datatype bar = Bar of Foo.foo''
end
