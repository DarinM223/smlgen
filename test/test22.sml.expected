structure Component1 =
struct
  datatype foo =
    Foo of {value: {opt1: foo option, opt2: (bar * foo) option} option}
  and bar =
    Bar of bar * foo
end
structure Foo = struct datatype foo = datatype Component1.foo end

structure Bar = struct datatype bar = datatype Component1.bar end